liblouis/compileTranslationTable.c

```diff
@@ -800,248 +790,235 @@ passFindCharacters(const FileInfo *file, widechar *instructions, int end,
 	// get the cell from the table, or if the cell is not defined yet, define it (without
 	// adding attributes)
 	dots = putDots(file, cell, table);
-	if (newRule->opcode >= CTO_Space && newRule->opcode < CTO_UpLow)
-		dots->definitionRule = newRuleOffset;
-	currentOffsetPtr = &dots->otherRules;
-	while (*currentOffsetPtr) {
-		currentRule = (TranslationTableRule *)&(*table)->ruleArea[*currentOffsetPtr];
-		if (newRule->charslen > currentRule->charslen || currentRule->dotslen == 0) break;
-		if (currentRule->opcode >= CTO_Space && currentRule->opcode < CTO_UpLow)
-			if (!(newRule->opcode >= CTO_Space && newRule->opcode < CTO_UpLow)) break;
-		currentOffsetPtr = &currentRule->dotsnext;
+	if (rule->opcode >= CTO_Space && rule->opcode < CTO_UpLow)
+		dots->definitionRule = ruleOffset;
+	TranslationTableOffset *otherRule = &dots->otherRules;
+	while (*otherRule) {
+		TranslationTableRule *r = (TranslationTableRule *)&(*table)->ruleArea[*otherRule];
+		if (rule->charslen > r->charslen || r->dotslen == 0) break;
+		if (r->opcode >= CTO_Space && r->opcode < CTO_UpLow)
+			if (!(rule->opcode >= CTO_Space && rule->opcode < CTO_UpLow)) break;
+		otherRule = &r->dotsnext;
 	}
-	newRule->dotsnext = *currentOffsetPtr;
-	*currentOffsetPtr = newRuleOffset;
+	rule->dotsnext = *otherRule;
+	*otherRule = ruleOffset;
 }
 
 static void
-addBackwardRuleWithMultipleCells(widechar *cells, int count,
-		TranslationTableOffset newRuleOffset, TranslationTableRule *newRule,
+addBackwardRuleWithMultipleCells(widechar *cells, int dotslen,
+		TranslationTableOffset ruleOffset, TranslationTableRule *rule,
 		TranslationTableHeader *table) {
-	/* direction = 1, newRule->dotslen > 1 */
-	TranslationTableRule *currentRule = NULL;
-	TranslationTableOffset *currentOffsetPtr =
-			&table->backRules[_lou_stringHash(cells, 0, NULL)];
-	if (newRule->opcode == CTO_SwapCc) return;
-	while (*currentOffsetPtr) {
-		int currentLength;
-		int newLength;
-		currentRule = (TranslationTableRule *)&table->ruleArea[*currentOffsetPtr];
-		currentLength = currentRule->dotslen + currentRule->charslen;
-		newLength = count + newRule->charslen;
-		if (newLength > currentLength) break;
-		if (currentLength == newLength)
-			if ((currentRule->opcode == CTO_Always) && (newRule->opcode != CTO_Always))
-				break;
-		currentOffsetPtr = &currentRule->dotsnext;
+	/* direction = 1, dotslen > 1 */
+	TranslationTableOffset *backRule = &table->backRules[_lou_stringHash(cells, 0, NULL)];
+	if (rule->opcode == CTO_SwapCc) return;
+	int ruleLength = dotslen + rule->charslen;
+	while (*backRule) {
+		TranslationTableRule *r = (TranslationTableRule *)&table->ruleArea[*backRule];
+		int rLength = r->dotslen + r->charslen;
+		if (ruleLength > rLength) break;
+		if (rLength == ruleLength)
+			if ((r->opcode == CTO_Always) && (rule->opcode != CTO_Always)) break;
+		backRule = &r->dotsnext;
 	}
-	newRule->dotsnext = *currentOffsetPtr;
-	*currentOffsetPtr = newRuleOffset;
+	rule->dotsnext = *backRule;
```

Comment:
Here you call it `backRule`. I see that there is some consistency :-).  For single chars/cells its called `otherRule` and for multiple chars/cells it's called `forRule`/`backRule`. As a casual code reader it is not clear to me whether there is a real difference between these four variables.